### PR TITLE
Implement v1alpha2 rig.dev Capsule spec

### PR DESCRIPTION
**Pull request by andersjohnsen**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,9 +58,9 @@ manifests: controller-gen ## 🪄 Generate k8s manifests
 		output:rbac:dir=deploy/kustomize/rbac \
 		output:webhook:dir=deploy/kustomize/webhook \
 		output:crd:dir=deploy/kustomize/crd/bases
-	echo '{{- if .Values.installCRDs }}' > deploy/charts/rig-operator/templates/crd.yaml
-	cat 'deploy/kustomize/crd/bases/rig.dev_capsules.yaml' >> deploy/charts/rig-operator/templates/crd.yaml
-	echo '{{- end }}' >> deploy/charts/rig-operator/templates/crd.yaml
+	# echo '{{- if .Values.installCRDs }}' > deploy/charts/rig-operator/templates/crd.yaml
+	# cat 'deploy/kustomize/crd/bases/rig.dev_capsules.yaml' >> deploy/charts/rig-operator/templates/crd.yaml
+	# echo '{{- end }}' >> deploy/charts/rig-operator/templates/crd.yaml
```
Comment (MatiasFrank) [inline Makefile:63]: Do we have an alternative to this for generating CRD spec?

File: pkg/api/v1alpha1/capsule_conversion.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+func (src *Capsule) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1alpha2.Capsule)
+
+	dst.ObjectMeta = src.ObjectMeta
+
+	dst.Spec.Args = src.Spec.Args
+	dst.Spec.Command = src.Spec.Command
+	dst.Spec.Image = src.Spec.Image
+	dst.Spec.NodeSelector = src.Spec.NodeSelector
```
Comment (MatiasFrank) [inline pkg/api/v1alpha1/capsule_conversion.go:20]: Should we deep copy the map?

File: Makefile (hunk 1 above)
Comment (blacksails) [reply Makefile:63]: we are already generating the CRD spec, this is generating a helm template based on the yaml which `controller-gen` outputs. We could maybe do something else, but templates for templates quickly get confusing xD

**Comment by blacksails** [comment]
Hope it wont be too much of a hassle to rebase with https://github.com/rigdev/rig/pull/259 landed. There is a small addition in the capsule spec in there which we should include in alpha2.

**Comment by andersjohnsen** [comment]
> Hope it wont be too much of a hassle to rebase with #259 landed. There is a small addition in the capsule spec in there which we should include in alpha2.

I'll merge them together, no worry :D

**Pull request merged**